integrations/pgvector/src/haystack_integrations/document_stores/pgvector/document_store.py

```diff
@@ -162,6 +163,13 @@ def to_dict(self) -> Dict[str, Any]:
             hnsw_ef_search=self.hnsw_ef_search,
         )
 
+    @classmethod
+    def from_dict(cls, init_parameters: Dict[str, Any]) -> "PgvectorDocumentStore":
+        conn_str_data = init_parameters["init_parameters"]["connection_string"]
+        conn_str = Secret.from_dict(conn_str_data) if conn_str_data is not None else None
+        init_parameters["init_parameters"]["connection_string"] = conn_str
+        return default_from_dict(cls, init_parameters)
```

Comment:
Not sure if this part can be simplified
(something like [this](https://github.com/deepset-ai/haystack-core-integrations/blob/9c069c948983a4b725a056c46cd347ab52349656/integrations/uptrain/src/haystack_integrations/components/evaluators/uptrain/evaluator.py#L157-158))

Reply:
nice! thanks, well spotted :) 👍🏽